packages/api-plugin-promotions/src/index.js

```diff
@@ -0,0 +1,45 @@
+import { createRequire } from "module";
+import { promotions, registerPluginHandlerForPromotions } from "./registration.js";
+import mutations from "./mutations/index.js";
+import startupPromotions from "./startup.js";
+import preStartupPromotions from "./preStartup.js";
+import { Promotion } from "./simpleSchemas.js";
+import actions from "./actions/index.js";
+import qualifiers from "./qualifiers/index.js";
+
+const require = createRequire(import.meta.url);
+const pkg = require("../package.json");
+
+/**
+ * @summary Import and call this function to add this plugin to your API.
+ * @param {Object} app The ReactionAPI instance
+ * @returns {undefined}
+ */
+export default async function register(app) {
+  await app.registerPlugin({
+    label: pkg.label,
+    name: pkg.name,
+    version: pkg.version,
+    collections: {
+      Promotions: {
+        name: "Promotions"
```

Comment:
The promotions, especially the implicit ones will be queried a lot and won't be updated often. That's why we should add indexes to some promotions' attributes. For example in the `getImplicitPromotions`, we are querying `enabled`, `type`, `startDate`, `endDate`. Also, each promotion implementation plugin should define it's own indexes, provided they are not already indexed in the root plugin.
 
The `coupons` plugin should index the `triggers.triggerKey`, `triggers.triggerParameters.couponCode` attributes.

Reply:
I'm not sure if `api-core` allows to add indexes to a collection that is already defined in another plugin.